src/views/Register.jsx

```diff
@@ -1,72 +1,134 @@
-import React from 'react';
-import Container from 'react-bootstrap/Container';
-import Form from 'react-bootstrap/Form';
-import Button from 'react-bootstrap/Button';
-import Row from 'react-bootstrap/Row';
-import Col from 'react-bootstrap/Col';
+import React, { useState } from "react";
+import axios from "axios";
+import Container from "react-bootstrap/Container";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import Row from "react-bootstrap/Row";
+import Col from "react-bootstrap/Col";
+import Alert from "react-bootstrap/Alert";
 
 /**
  * Implement the user registration functionality.
- * 
+ *
  * You may limit your model to:
  * {
  *  firstName: string,
  *  lastName: string,
  *  username: string,
  *  password: string,
  * }
- * 
+ *
  * When the form is saved, send your payload to the following endpoint:
  * POST https://api-demo.kingslandtesting.com/user/register
  */
 const Register = () => {
+  const [data, setData] = useState({
+    firstName: "",
+    lastName: "",
+    username: "",
+    password: "",
+  });
+  const [registered, setRegistered] = useState(false);
+
+  const handleChange = (e) => {
+    setData({ ...data, [e.target.name]: e.target.value });
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+
+    const userData = {
+      firstName: data.firstName,
+      lastName: data.lastName,
+      username: data.username,
+      password: data.password,
+    };
+
+    await axios
+      .post("https://api-demo.kingslandtesting.com/user/register", userData, {
+        "Content-Type": "application/json",
+      })
+      .then((response) => {
+        console.log(response.data);
+        if (response.data) setRegistered(true);
+      })
+      .catch((error) => console.log(error));
```

Comment:
Error is not properly handled. No prompts are shown to the user if the username is duplicate or if there are registration errors.